src/main/scala/com/atomist/tree/content/text/microgrammar/MatcherMicrogrammar.scala

```diff
@@ -23,10 +23,9 @@ class MatcherMicrogrammar(val matcher: Matcher, val name: String = "MySpecialMic
     processedNodes
   }
 
-  def strictMatch(input: CharSequence, l: Option[MatchListener] = None): MutableContainerTreeNode = {
-    val nodes = findMatchesInternal(input, l)
-    require(nodes.size == 1, s"Expected 1 result, not ${nodes.size}")
-    outputNode(input)(nodes.head._1)
+  def strictMatch(input: CharSequence, l: Option[MatchListener] = None): Either[DismatchReport, MutableContainerTreeNode] = {
+    val result = matcher.matchPrefix(InputState(input))
+    result.right.map(matched => outputNode(input)(matched, OffsetInputPosition(0)))
```

Comment:
Yep. This is the best part. I was considering deleting this function (only used in tests) but instead it has become useful!
It used to throw an exception if it didn't match exactly once. woo.
The -purpose- of the method is to check that something you expect to match, definitely does.
now it fulfills its purpose better: it gives you the matched node, or it gives you the dismatchReport. (currently this is the only way to get a DismatchReport.) So tests that use this method can print something useful when they fail!